pkg/metrics/metrics.go

```diff
@@ -13,38 +13,49 @@ func init() {
 	m = kybermetric.Meter()
 }
 
-func RecordFloat64Histogram(ctx context.Context, name string, value float64) error {
+func RecordFloat64Histogram(ctx context.Context, name string, value float64, options ...metric.RecordOption) error {
 	hist, err := m.Float64Histogram(name)
 	if err != nil {
 		return err
 	}
-	hist.Record(ctx, value)
+	hist.Record(ctx, value, options...)
 	return nil
 }
 
-func RecordFloat64Gause(ctx context.Context, name string, value float64) error {
+func RecordFloat64Gause(ctx context.Context, name string, value float64, options ...metric.ObserveOption) error {
 	_, err := m.Float64ObservableGauge(name,
 		metric.WithFloat64Callback(func(ctx context.Context, fo metric.Float64Observer) error {
-			fo.Observe(value)
+			fo.Observe(value, options...)
 			return nil
 		}))
```

Comment:
This is not a good implementation for a callback function. The Observable Gauge will call the register function at intervals, so if we set a fixed 'value,' it will just send the same 'value' repeatedly without reflecting real-time changes.

Reply:
yes, I know. I will call this func every time value change, so updated value will be send.

Reply:
It mean I will update callback every time value change, not much performance for that

Reply:
Gauge observe function will call function callback at intervals, So even if you don't change the value, it still send the value to the metric.

Reply:
removed 😄